common/src/main/java/org/mvndaemon/mvnd/common/OsUtils.java

```diff
@@ -101,6 +102,29 @@ public static long findProcessRssInKb(long pid) {
         }
     }
 
+    public static String findJavaHomeFromPATH() {
+        final String basename = Os.current().isUnixLike() ? "java" : "java.exe";
+        final String PATH = System.getenv("PATH");
+        if (null == PATH) {
+            return null;
+        }
+        for (final String dirname : PATH.split(File.pathSeparator)) {
+            final File java = new File(dirname, basename);
+            if (java.isFile() && java.canExecute()) {
+                final String[] cmd = { java.getAbsolutePath(), "-XshowSettings:properties", "-version" };
+                final List<String> output = new ArrayList<String>(1);
```

Comment:
The size of `1` does not make much sense here.  The output will most probably include much more lines.

Reply:
In the latest commit, `File` is removed.

Reply:
Ah, I missed this commit, thx !